app/Http/Controllers/ConnectionController.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\User;
+use App\Connection;
+use Illuminate\Support\Facades\Auth;
+
+class ConnectionController extends Controller
+{
+    public function show($id)
+    {
+        $user = User::findOrFail($id);
+        $incoming = Connection::getIncomingConnections($user);
```

Comment:
It's much cleaner than before with these static methods. But, we can still refactor this further.

Think about the operations going on here.

We have a `User`, then we call a static method on a different class and pass IN the user.

Perhaps we should invert the dependencies, and make our code more expressive?

Imagine that instead it was `$user->incomingConnections()` and `$user->outgoingConnections()`;

Under the hood those can still just be wrapping the static method `Connection::getIncomingConnections($this)`, but it reduces the surface areas of classes we're directly interacting with, and properly shows the relationship between the user and their connections.

Reply:
BONUS THOUGHT EXERCISE: What kind of relationships between what entities are we actually modelling with these connections and have we modelled it in the best way possible?

Reply:
Interesting. I think the answer is separating the incoming and outgoing tables (actually, these might be able to be eliminated), as ultimately they are one-to-many for outgoing connections. Be interested to hear your intended answer.